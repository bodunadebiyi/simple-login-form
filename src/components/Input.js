import React from 'react';
import injectSheet from 'react-jss';
import PropTypes from 'prop-types';

const styles = {
  wrapper: {
    textAlign: 'left',
    marginBottom: 20,
  },
  label: {
    display: 'block',
    fontSize: 11,
    marginBottom: 5,
  },
  input: {
    display: 'block',
    width: '100%',
    padding: 10,
    boxSizing: 'border-box',
    outline: 'none',
    borderStyle: 'solid',
    borderWidth: 1,
    borderColor: '#999',
    transition: 'all ease .2s',
    '&:focus': {
      borderColor: 'green',
    }
  }
};

const Input = React.memo(({
  classes,
  label,
  inputProps
}) => {
  return (
    <div className={classes.wrapper}>
      {label && <label
        htmlFor={inputProps.id}
        className={classes.label}
      >
        { label }
      </label>}
      <input
        className={classes.input} 
        {...inputProps} />
    </div>
  )
});

Input.defaultProps = {
  inputProps: {
    type: 'text'
  }
}

Input.propTypes = {
  label: PropTypes.string,
  inputProps: PropTypes.object,
}

export default injectSheet(styles)(Input);