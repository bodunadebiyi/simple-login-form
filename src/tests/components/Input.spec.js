import React from 'react';
import { shallow } from 'enzyme';
import Input from '../../components/Input';

describe('Input Component', () => {
  describe('when component is mounted', () => {
    it('renders without crashing', () => {
      const component = shallow(<Input />)
      expect(component.exists()).toBe(true);
    });

    it('matches snapshot', () => {
      const props = {
        label: 'My Input',
        inputProps: {
          id: 'my_input',
          type: 'text'
        }
      }
      
      const component = shallow(<Input {...props} />)
      expect(component).toMatchSnapshot();
    });
  });

  describe('when label prop is present', () => {
    it('expects label html component to be present', () => {
      const props = { label: 'my label' };
      const component = shallow(<Input {...props} />);
      const label = component.dive().find('label');

      expect(label.exists()).toBe(true);
    });
  });

  describe('when label prop is NOT present', () => {
    it('expects label html component to NOT be present', () => {
      const component = shallow(<Input />);
      const label = component.dive().find('label');

      expect(label.exists()).toBe(false);
    });
  });

  describe('when input is changed', () => {
    it('expects onChange handler to be called', () => {
      const props = {
        label: 'email',
        inputProps: {
          onChange: jest.fn(),
          name: 'email',
          type: 'text'
        }
      };

      const component = shallow(<Input {...props} />);
      component.dive().find('input').simulate('change');

      expect(props.inputProps.onChange).toHaveBeenCalled();
    });
  });
});
