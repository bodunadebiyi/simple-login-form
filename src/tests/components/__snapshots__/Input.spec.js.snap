// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component when component is mounted matches snapshot 1`] = `ShallowWrapper {}`;
